### PR TITLE
[ST] Refactor - Rename systemtest class Constants to TestConstants

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change
- Refactoring

### Description

In the STs like AbstractUpgradeST, we encountered importing more constant classes named Constants. Specifically, we were using the same class from different modules, which might lead to potential conflicts and confusion.
To address this, this PR includes renaming our Constants class in systemtest module to something more descriptive and less likely to cause naming conflicts with other classes. This renaming should help maintain code clarity and avoids unintended conflicts, ensuring that our code won't be confusing.

Additionally, this PR also adjusts all of these import statements - all of its ocurences.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=UserST

**Review by see-quick — APPROVED** [review]
Assuming the test is going to pass.  LGTM

**Review by Frawless — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/TestConstants.java [hunk 1]
```diff
@@ -174,19 +174,19 @@ public interface Constants {
...
+    String PATH_TO_KAFKA_CRUISE_CONTROL_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/cruise-control/kafka-cruise-control.yaml";
+    String PATH_TO_KAFKA_EPHEMERAL_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/kafka/kafka-ephemeral.yaml";
+    String PATH_TO_KAFKA_PERSISTENT_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/kafka/kafka-persistent.yaml";
+    String PATH_TO_KAFKA_CRUISE_CONTROL_METRICS_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/metrics/kafka-cruise-control-metrics.yaml";
+    String PATH_TO_KAFKA_TOPIC_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/topic/kafka-topic.yaml";
+    String PATH_TO_KAFKA_CONNECTOR_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/connect/source-connector.yaml";
+    String PATH_TO_KAFKA_MIRROR_MAKER_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/mirror-maker/kafka-mirror-maker.yaml";
+    String PATH_TO_KAFKA_MIRROR_MAKER_2_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/mirror-maker/kafka-mirror-maker-2.yaml";
+    String PATH_TO_KAFKA_MIRROR_MAKER_2_METRICS_CONFIG = TestConstants.PATH_TO_PACKAGING_EXAMPLES + "/metrics/kafka-mirror-maker-2-metrics.yaml";
```
[the remark this answers is not in the document]
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/TestConstants.java:189]: Thanks! Thats a good catch! sorry for that.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public abstract class AbstractST implements TestSeparator {
 
     public static Random rng = new Random();
 
-    public static final int MESSAGE_COUNT = Constants.MESSAGE_COUNT;
+    public static final int MESSAGE_COUNT = TestConstants.MESSAGE_COUNT;
     public static final String USER_NAME = KafkaUserUtils.generateRandomNameOfKafkaUser();
     public static final String TOPIC_NAME = KafkaTopicUtils.generateRandomNameOfTopic();
```
[the remark this answers is not in the document]
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:98]: Can do that in another PR! Thanks

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java [hunk 3]
```diff
@@ -104,8 +104,8 @@ public static void createFileSinkConnector(String namespaceName, String podName,
     }
 
     public static void waitForConnectorsTaskMaxChange(String namespaceName, String connectorName, int taskMax) {
-        TestUtils.waitFor("KafkaConnector taskMax change", Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.GLOBAL_TIMEOUT,
-            () -> (KafkaConnectorResource.kafkaConnectorClient().inNamespace(namespaceName)
+        TestUtils.waitFor("KafkaConnector taskMax change", TestConstants.POLL_INTERVAL_FOR_RESOURCE_READINESS, TestConstants.GLOBAL_TIMEOUT,
+                          () -> (KafkaConnectorResource.kafkaConnectorClient().inNamespace(namespaceName)
```
[the remark this answers is not in the document]
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java:108]: Will change that, thanks

**Comment by jankalinic** [comment]
Tests on azp buildId=155378 finished with one error -> testKafkaDoesNotRollsWhenTopicIsUnderReplicated

**Comment by jankalinic** [comment]
I think this PR is ready to be merged as AZP showed only unrelated test ->testKafkaDoesNotRollsWhenTopicIsUnderReplicated failing.

**Comment by scholzj** [comment]
@im-konge Do you have any more comments on this? Or should we proceed and merge this? You seem to have commented but did not approve.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java [hunk 4]
```diff
@@ -73,11 +73,11 @@ void testScramShaAuthWithWeirdUsername(ExtensionContext extensionContext) {
...
         KafkaBridgeSpec bridgeSpec = new KafkaBridgeSpecBuilder()
             .withNewKafkaClientAuthenticationScramSha512()
-                .withUsername(weirdUserName)
-                .withPasswordSecret(passwordSecret)
+            .withUsername(weirdUserName)
+            .withPasswordSecret(passwordSecret)
             .endKafkaClientAuthenticationScramSha512()
             .withNewTls()
-                .withTrustedCertificates(certSecret)
+            .withTrustedCertificates(certSecret)
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java:80]: You are changing indentation in the whole class, why?

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 5]
```diff
@@ -282,15 +282,15 @@ void testConnectLogSetting(ExtensionContext extensionContext) {
...
-                .endInlineLogging()
-                .withNewJvmOptions()
-                    .withGcLoggingEnabled(true)
-                .endJvmOptions()
+            .withNewInlineLogging()
+            .withLoggers(CONNECT_LOGGERS)
+            .endInlineLogging()
+            .withNewJvmOptions()
+            .withGcLoggingEnabled(true)
+            .endJvmOptions()
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:293]: Here as well

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 6]
```diff
@@ -144,55 +144,55 @@ void testJSONFormatLogging(ExtensionContext extensionContext) {
...
 
         String originalCoLoggers = kubeClient().getClient().configMaps()
             .inNamespace(clusterOperator.getDeploymentNamespace()).withName(configMapCOName).get().getData().get("log4j2.properties");
 
         ConfigMap configMapKafka = new ConfigMapBuilder()
             .withNewMetadata()
-                .withName(configMapKafkaName)
-                .withNamespace(namespaceName)
+            .withName(configMapKafkaName)
+            .withNamespace(namespaceName)
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:155]: Here as well

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 7]
```diff
@@ -109,34 +109,28 @@
...
+ * 5. - Deploy scraper Pods in namespace {@namespaceFirst} and {@namespaceSecond} for collecting metrics from Strimzi pods
+ * - Scraper Pods are deployed
+ * 6. - Create KafkaUsers and KafkaTopics
+ * - All KafkaUsers and KafkaTopics are Ready
+ * 7. - Setup NetworkPolicies to grant access to Operator Pods and KafkaExporter
+ * - NetworkPolicies created
+ * 8. - Create collectors for Cluster Operator, Kafka, KafkaExporter, and Zookeeper (Non-KRaft)
+ * - Metrics collected in collectors structs
+ * @afterAll 1. - Common cleaning of all resources created by this test class
+ * - All resources deleted.
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:133]: Why are you changing the lines etc.?

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 8]
```diff
@@ -645,30 +595,30 @@ void testKafkaMetricsSettings() {
...
-                    .withName("external-metrics-cm")
-                    .withNamespace(namespaceSecond)
-                .endMetadata()
-                .build();
+            .withData(Collections.singletonMap(TestConstants.METRICS_CONFIG_YAML_NAME, metricsConfigYaml))
+            .withNewMetadata()
+            .withName("external-metrics-cm")
+            .withNamespace(namespaceSecond)
+            .endMetadata()
+            .build();
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:608]: Again indent

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java [hunk 9]
```diff
@@ -150,17 +150,17 @@ void testCreateDeleteCreate(ExtensionContext extensionContext) throws Interrupte
...
-        String adminClientPodName = kubeClient().listPods(testStorage.getNamespaceName(), Constants.ADMIN_CLIENT_LABEL_SELECTOR).get(0).getMetadata().getName();
+        String adminClientPodName = kubeClient().listPods(testStorage.getNamespaceName(), TestConstants.ADMIN_CLIENT_LABEL_SELECTOR).get(0).getMetadata().getName();
 
         AdminClient adminClient = new AdminClient(testStorage.getNamespaceName(), adminClientPodName);
         adminClient.configureFromEnv();
 
         resourceManager.createResourceWithWait(extensionContext, KafkaTopicTemplates.topic(KAFKA_CLUSTER_NAME, testStorage.getTopicName(), Environment.TEST_SUITE_NAMESPACE)
             .editMetadata()
-                .withNamespace(Environment.TEST_SUITE_NAMESPACE)
+            .withNamespace(Environment.TEST_SUITE_NAMESPACE)
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:160]: Here as well

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java [hunk 10]
```diff
@@ -259,25 +259,21 @@ void testRecoveryDuringKafkaRollingUpdate(ExtensionContext extensionContext) {
...
+     * - Cluster scales to 5 replicas and volumes as such
+     * 4. - Deploy KafkaTopic with 4 replicas and new clients which will target this KafkaTopic and the first one KafkaTopic
+     * - Topic is deployed and ready, clients successfully communicate with respective KafkaTopics
+     * 5. - Scale down Kafka Cluster back from 5 to 3 replicas
+     * - Cluster scales down to 3 replicas and volumes as such
+     * 6. - Deploy new KafkaTopic and new clients which will target this KafkaTopic, also do the same for the first KafkaTopic
+     * - New KafkaTopic is created and ready, clients successfully communicate with respective KafkaTopics
+     * @usecase - kafka
+     * - scale-up
+     * - scale-down
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java:276]: Why is it changed? It's completely un-related to the PR description

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java [hunk 11]
```diff
@@ -291,21 +287,21 @@ void testKafkaScaleUpScaleDown(ExtensionContext extensionContext) {
         final String topicNameScaledBackDown = testStorage.getTopicName() + "-scaled-down";
 
         resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3, 3)
-            .editMetadata()
+                .editMetadata()
                 .addToAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_SKIP_BROKER_SCALEDOWN_CHECK, "true"))
-            .endMetadata()
-            .editSpec()
+                .endMetadata()
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java:292]: Indent

**Review by im-konge — CHANGES_REQUESTED** [review]
You are changing indentations and test descriptions even if it's not the main target of the PR. I left few comments, please check the whole classes and fix the things you broke in the last commit. Thanks

**Review by im-konge — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**